### ISSUE TITLE
[Bug] `LinkTo` with incomplete model failing in rendering tests

### ISSUE THREAD

**Issue by Turbo87** [emberjs/ember.js#19364]
### 🐞 Describe the Bug

We have a few components that use `LinkTo` components but are only used on subroutes with dynamic segments and have not explicitly added `@model` (e.g. only used within `/organizations/:org_id`). This works fine in production and up until Ember 3.24 this also worked fine in application and rendering tests.

With 3.24 Ember now automatically starts the routing layer whenever something on the `router` service is accessed, and now the `LinkTo` component is complaining that we did not pass the necessary models when any previous thing in the template for example accesses `router.currentRouteName`.

### 🔬 Minimal Reproduction
Describe steps to reproduce. If possible, please, share a link with a minimal reproduction.

```js
// router.js

Router.map(function() {
  this.route('parent', { path: 'parent/:parent_id' }, function () {
    this.route('child');
  });
});
```

```js
test('broken', async function(assert) {
  // access a property on the router
  this.owner.lookup('service:router').currentRouteName;

  // render a LinkTo with incomplete model
  await render(hbs`<LinkTo @route="parent.child">Link</LinkTo>`);
});
```

### 😕 Actual Behavior

> Assertion Failed: You attempted to generate a link for the "parent.child" route, but did not pass the models required for generating its dynamic segments. Cannot read property 'shouldSupercede' of undefined

### 🤔 Expected Behavior

> The `LinkTo` should render without an `href` like it did prior to Ember 3.24, or the `router.currentRouteName` access should not have the side-effect of automatically starting the routing layer, which is causing these issues.

### 🌍 Environment

- Ember: 3.24

### ➕ Additional Context

somewhat related: 
- https://github.com/emberjs/ember.js/pull/19080
- https://github.com/buschtoens/ember-link/pull/518
- https://github.com/buschtoens/ember-link/pull/507

/cc @rwjblue @xg-wang

**Comment by xg-wang** [comment]
Sorry for the delay, I won't have much time until the weekend to look into the fix

**Comment by rwjblue** [comment]
Chatted about this with @xg-wang earlier, I think what we should do is to change this guard:

https://github.com/emberjs/ember.js/blob/e25c0f258603fbfd57a47c14c00f69b93805ec46/packages/%40ember/-internals/routing/lib/services/routing.ts#L57-L60

To be something like:

```ts
    let router = this.router;
    // return early when the router microlib is not present, which is the case for {{link-to}} in integration tests
    if (!router._hasBeganInitialRouting) {
      return;
    }
```

Then update:

https://github.com/emberjs/ember.js/blob/e25c0f258603fbfd57a47c14c00f69b93805ec46/packages/%40ember/-internals/routing/lib/system/router.ts#L407-L418

To set that flag.

**Comment by rwjblue** [comment]
Fixed by #19387, released in 3.25.1 and 3.24.2.
